Automattic/wp-calypso
# Marketing Connections: Fix tour position (updated seenTarget logic)

#### Changes proposed in this Pull Request

This is a fixed version of #44329. After merging @allilevine and I discovered that the tour wouldn't display. @creativecoder found this could be consistently reproduced by clearing Index DB. 

From what I can tell, the initialisation of the state caused by clearing index DB caused the order of asynchronously loaded code and the subsequent rendering of components to change, which threw up a bug with the MutationObserver. The tour step has an `onTargetDisappear` property, which is a callback used when the target element is no longer found in the DOM. The problem was that the logic wasn't checking that the target had ever appeared in the DOM and so was causing the tour to quit before it had been displayed.

#### Testing instructions

The instructions are generally the same as #44329. 

* Go to `/marketing/connections/[YOUR TEST SITE]?tour=marketingConnectionsTour`
* Without this change, the tour should display incorrectly at the top left of the screen. You may have to delete the Calypso database from Index DB to see this bug now
* With this change, the tour should appear once the connection statuses have loaded and as the screen changes it should adjust its position.
* Still with the change applied, delete the Calypso Index DB database, and check that the tour still functions as expected.

The bug looks like this:

<img width="1554" alt="image" src="https://user-images.githubusercontent.com/96462/88866321-43f33000-d202-11ea-8251-e4e7c4681fa9.png">

With the change applied:

<img width="1554" alt="image" src="https://user-images.githubusercontent.com/96462/88866387-7d2ba000-d202-11ea-97f3-48b2c711e04c.png">

Fixes #44194
